src-tauri/src/main.rs

```diff
@@ -5,24 +5,29 @@ pub mod api;
 pub mod models;
 
 use crate::api::weather_api::fetch_basic_weather_data;
-use crate::models::weather_response::CurrentWeatherResponse;
+use crate::models::weather_response::{WeatherData, Location};
 
 #[tauri::command]
-async fn get_wmo_code_description(code: i32) -> String {
-    models::wmo_code::create_wmo_code_map()
-    .get(&code)
-    .unwrap_or(&String::from("Error retrieving WMO code description"))
-    .to_owned()
-}
-
-#[tauri::command]
-async fn get_weather_data(_location: &str) -> Result<CurrentWeatherResponse, ()> {
+async fn get_weather_data(_location: &str) -> Result<WeatherData, ()> {
     let Ok(geocoding_result) = api::geocoding::get_coordinates(_location).await else {
         return Err(());
     };
-    let data = fetch_basic_weather_data(geocoding_result).await;
+    let data = fetch_basic_weather_data(&geocoding_result).await;
     match data {
-        Ok(response) => Ok(response),
+        Ok(response) => {
+            let weather_data = WeatherData {
+                location: Location {
+                    name: geocoding_result.name,
+                    country: geocoding_result.country
+                },
+                wmo_code_description: models::wmo_code::get_wmo_code_description(response.current.weathercode),
+                current_units: response.current_units,
+                current: response.current,
+                daily_units: response.daily_units,
+                daily: response.daily
+            };
```

Comment:
For the `From` impl, is what you have in mind that it can build the `WeatherData` struct from a `WeatherApiResponse` and `GeocodingResult`?

So `WeatherData.from(weather_api_response, geocoding_result)`

Or do you mean something else?

Reply:
Yes, something like that, although on further thought, `From<T>` is not the right trait for this since it can only take in one input value `T` (it's intended for direct value to value conversion). 

It would be better if you would create a `new()` function, so something like this:

```rust
impl WeatherData {
    fn new(args) {
        // body
    }
}
```